builders/build/eth-api/dev-env/foundry.md

```diff
@@ -143,11 +147,45 @@ forge build
     src/MyToken.sol:MyToken
     ```
 
-几分钟后，合约完成部署，您将在终端看到地址。
+部署合约后几秒钟，您应该会在终端看到合约地址。
+
+--8<-- 'code/builders/build/eth-api/dev-env/foundry/terminal/deploy.md'
+
+恭喜！您的合约已部署成功！请保存该地址，您将在下一步中使用它与该合约实例进行交互。
+
+### 使用Solidity脚本部署 {: #deploying-via-solidity-scripting }  
+
+Solidity脚本是一种比 [`forge create`](#deploying-the-contract) 更强大、更灵活的合约部署方式。 编写Solidity脚本与编写普通的Solidity智能合约相同，但您永远不会部署此脚本。
+
+您可以使用各种参数定制`forge script`的行为。除了本地模拟（这是每次运行的必要部分）之外，所有组件都是可选的。`forge script`命令将按照以下顺序尝试执行所有步骤：
+
+1. **Local simulation** - 在本地EVM中模拟交易
+2. **Onchain simulation** - 通过提供的RPC URL模拟交易
+3. **Broadcasting** - 当提供 `--broadcast` 标志并且模拟成功时，交易将被发送
+4. **Verification** - 当提供 `--verify` 标志和有效API密钥时，进行基于API的智能合约验证
+
+现在开始编写脚本。在脚本文件夹中，创建一个名为 `MyToken.s.sol` 的文件。复制并粘贴以下文件的内容：
+
+```solidity
+--8<-- 'code/builders/build/eth-api/dev-env/foundry/MyToken-script.sol'
+```
+
+!!! 请记住
+    请记住永远不要像上面那样将生产私钥存储在文件中。 这个例子仅用于演示目的。
+
+请注意，即使上述脚本不会被部署，它仍需要符合典型Solidity合约的所有格式，例如pragma语句。
+
+您可以使用以下命令部署 `MyToken.sol` 合约。请记住它将按顺序执行所有相关步骤。对于此示例，Foundry将首先尝试本地模拟和利用提供的 RPC进行模拟，然后才部署合约。如果任何模拟失败，Foundry都不会继续部署。
+
+```bash
+forge script script/MyToken.s.sol --rpc-url https://rpc.api.moonbase.moonbeam.network --broadcast
+```
+
+如果您的脚本执行成功，您的终端应展示一下内容：
 
-![Foundry Contract Deploy](/images/builders/build/eth-api/dev-env/foundry/foundry-2.webp)
+--8<-- 'code/builders/build/eth-api/dev-env/foundry/terminal/script.md'
 
-恭喜您，您的合约已上线！保存地址，用于后续合约实例交互。
+这样就成了。有关使用Foundry进行Solidity脚本编写的更多信息，请查看[Foundry文档](https://book.getfoundry.sh/tutorials/solidity-scripting){target=_blank}。
```

Comment:
```suggestion
这样就成了。有关使用Foundry进行Solidity脚本编写的更多信息，请查看[Foundry文档](https://book.getfoundry.sh/tutorials/solidity-scripting){target=\_blank}。
```

Reply:
Closest I could get, but on line 194, can you please change `cast` to `Cast` with a capital C

and also, please change all instances of `YOUR_CONTRACT_ADDRESS` to `INSERT_YOUR_CONTRACT_ADDRESS` to follow convention